### PR TITLE
have more stringent javac options

### PR DESCRIPTION
**Pull request by stephane-airbyte**
This is a no-op in term of features or bugs.

I'm only trying to use as many compiler warnings as possible. They're cheap and can help with a lot of bugs.
Unfortunately, our CDK is currently packaged in a way that causes warnings in any package that uses it. I fixed that by making swagger an api-level dependency (which means it'll be exported to projects that depend on airbyte-cdk), but all projects that depend on it will have a warning, which means we can't turn those into errors.
So I'm releasing a new version of the CDK number 0.3.0, and enabling -Werror for any component that depends on a newer CDK version.

For the projects that don't depend on CDK, I'm disabling the errors that are present, rather than making a sweeping change across all projects. I'll try to fix as many of those as possible, and remove the special cases.

An extra special case is that of our performance harnesses. They use very old version of our packages, and simply don't compile when migrated to the latest versions. So for those I had to special-disable all errors.

### CODE REVIEW

File: gradle.properties
Comment (stephane-airbyte) [inline gradle.properties:8]: this was necessary because the new checks add to the memory consumption :(

**Comment by stephane-airbyte** [comment]
@aaronsteers Wanted to add you to make sure you were OK with the move to CDK 0.3

File: buildSrc/src/main/groovy/airbyte-java-connector.gradle
Comment (postamar) [inline buildSrc/src/main/groovy/airbyte-java-connector.gradle:61]: Can we do a regex match instead?
`def hasKnownWarnings = (cdkVersionRequired =~ "0\\.[0-2]\\.\\d+")` or whatever

File: build.gradle
Comment (postamar) [inline build.gradle:5]: Is this import deliberate? Looks like it might be detritus.

File: build.gradle
Comment (postamar) [inline build.gradle:14]: What does this do?

File: airbyte-integrations/connectors/source-postgres/build.gradle
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/build.gradle:11]: Do we want to file an issue to try to remove this special casing by getting the linter to pass?

File: airbyte-cdk/java/airbyte-cdk/core/build.gradle
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/build.gradle:72]: Thanks. In case you haven't noticed, the dependencies need to be cleaned up a bit. I don't see us doing this any other way than little piece by little piece. Every little bit helps.

File: airbyte-integrations/connectors/source-postgres/build.gradle
Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/build.gradle:11]: Same comment applies to the CDK. This isn't a rethorical question.

**Review by aaronsteers — APPROVED** [review]
My review and approval is based on the PR description and a quick scan through the code diff and comments.

I'll defer to @postamar for a more detailed code review, but for my part this looks like a great iteration in the right direction. 🙏

File: build.gradle
Comment (stephane-airbyte) [reply build.gradle:14]: this and above is a remnant of me trying to parse the metadata.yaml to only apply the extra checks to certified connectors. Instead, I'll be using the CDK version, which is simpler but means that stuff will start breaking whenever we try to upgrade the CDK version to 0.3.x

**Comment by stephane-airbyte** [comment]
@postamar can you take another look. I fixed a bunch of warnings in the certified connectors. I believe that's all of them!

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/state/StateManager.java
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/source/relationaldb/state/StateManager.java:25]: the generics are not used anywhere, so just removing the

File: airbyte-cdk/java/airbyte-cdk/acceptance-test-harness/build.gradle
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/acceptance-test-harness/build.gradle:28]: this is necessary for some weird versionning issues

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryGcsOperations.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryGcsOperations.java:163]: the overloaded method is also deprecated, so I'm marking this one as well

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordConsumer.java:34]: The problem here is that the throw of AirbyteMessageConsumer throws an Exception, which could be an InterruptedException, and that triggers a warning

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordStandardConsumer.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryRecordStandardConsumer.java:23]: The close of AsyncStreamConsumer throws an Exception

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java:149]: SnwoflakeColumnDefinition.isNullable is deprecated.

File: airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeV1V2Migrator.java:91]: NamingConventionTransformer.getRawTableName is deprecated

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java:166]: nasty one here. stats.get returns an Object by default. There is a get(String, Class<T>) that returns a T, but that doesn't work with a parameterized class, so no matter what, there's a suppresswarnings here :(

File: airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mongodb-v2/src/main/java/io/airbyte/integrations/source/mongodb/MongoUtil.java:259]: same as above

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java:47]: cleaning up whitespaces that would get removed by the compiler

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/MySqlQueryUtils.java:86]: Collections.EMPTY_SET is a Set, while Collections.emptySet() is a Set<>

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadHandler.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadHandler.java:52]: the implementation is always used, and implements a generic type. So we're getting rid of a rawtype warning

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java:42]: close throws an exception

File: airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/source-mysql/src/main/java/io/airbyte/integrations/source/mysql/initialsync/MySqlInitialLoadRecordIterator.java:48]: raw type

File: airbyte-integrations/connectors/source-postgres/build.gradle
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/source-postgres/build.gradle:11]: I was going to do it without a new issue, but I can do that

File: buildSrc/src/main/groovy/airbyte-java-connector.gradle
Comment (stephane-airbyte) [reply buildSrc/src/main/groovy/airbyte-java-connector.gradle:61]: I tend to find regexp harder to read, but if you feel strongly about it, I'll do the change

File: build.gradle
Comment (postamar) [reply build.gradle]: This CDK version approach sounds rather brittle, in the meantime why not hardcode a whitelist of certified connectors? Of course it will grow stale, but that's not so bad, because we never downgrade connectors from certified status to non-certified. Until we certify another connector we should be able to come up with a decent way to parse these files. Feel free to create an issue to that effect and assign it to me.

File: buildSrc/src/main/groovy/airbyte-java-connector.gradle
Comment (postamar) [reply buildSrc/src/main/groovy/airbyte-java-connector.gradle:61]: See comment below: I'd rather you maintain a hardcoded whitelist of connectors and check whether `project.name` is among them.

File: build.gradle
Comment (stephane-airbyte) [reply build.gradle]: I just realized there's a couple more reasons why I went that road. I initially did parse the metadata.xml, and things were working but:
1) we would need to split this change into several: 1 to push the api exports into the new CDK version, then one to update all certified connectors to the latest CDK version, and finally one to enable the compiler checks. This would be necessary because the current CDK is missing package exports, which means all connectors have warnings.
2) I was also thinking that it'd be nice to be more stringent with new connectors, certified or not. The CDK-version handling gives us that. If you still prefer using the certified flag, I'll create the 3  required PRs

**Review by postamar — APPROVED** [review]
LGTM modulo my request to use a project name whitelist to determine linter strictness, rather than checking the CDK version.

File: airbyte-integrations/connectors/source-postgres/build.gradle
Comment (postamar) [reply airbyte-integrations/connectors/source-postgres/build.gradle:11]: Whichever works best for you!

File: build.gradle
Comment (stephane-airbyte) [reply build.gradle]: Alternatively, I can just disable the -Werror for all the non-certified connectors, this way new connectors get the more stringent checks, and everything is simpler. That requires me to edit all the build.gradle, but I don't see that as a big deal

Comment (postamar) [reply build.gradle]: Oh, I like that idea!

**Review by rodireich — APPROVED** [review]

**Pull request closed without merging**
